src/index/base.cpp

```diff
@@ -33,6 +33,7 @@ constexpr auto SYNC_LOCATOR_WRITE_INTERVAL{30s};
 template <typename... Args>
 void BaseIndex::FatalErrorf(const char* fmt, const Args&... args)
 {
+    Interrupt(); // Cancel the sync thread
```

Comment:
In 5e2b1ea1:

The description says that this commit does not change behavior but this `Interrupt()` call here seems like a behavior change.